frontend/src/metabase/visualizations/echarts/cartesian/model/dataset.unit.spec.ts

```diff
@@ -178,4 +186,59 @@ describe("dataset transform functions", () => {
       expect(result).toEqual([]);
     });
   });
+
+  describe("replaceValues", () => {
+    it("should replace missing values with zeros according to the replacer function", () => {
+      const dataset: Record<DataKey, RowValue>[] = [
+        { key1: null, key2: 2 },
+        { key1: 3, key2: null },
+      ];
+      const replacer = (dataKey: string, value: RowValue) => {
+        if (dataKey === "key1") {
+          return value;
+        }
+        return value ?? 0;
+      };
+
+      const result = replaceValues(dataset, replacer);
+
+      expect(result).toEqual([
+        { key1: null, key2: 2 },
+        { key1: 3, key2: 0 },
+      ]);
+    });
+  });
+
+  describe("getNullReplacerFunction", () => {
+    it("should create a replacer function that replaces null values with zeros for specified series only", () => {
+      const createMockSeriesModel = (
```

Comment:
Should this be in a shared location so we can use it for other tests (e.g. `echarts/cartesian/model/test-utils.ts`?) Feel free to leave it here for now though since we aren't using it elsewhere yet, but maybe in a later PR we can move it